libtenzir/builtins/contexts/geoip.cpp

```diff
@@ -531,9 +552,59 @@ struct v1_loader : public context_loader {
   }
 };
 
+struct v2_loader : public context_loader {
+  explicit v2_loader(record global_config)
+    : global_config_{std::move(global_config)} {
+  }
+
+  auto version() const -> int {
+    return 2;
+  }
+
+  auto load(chunk_ptr serialized) const
+    -> caf::expected<std::unique_ptr<context>> {
+    const auto* cache_dir
+      = get_if<std::string>(&global_config_, "tenzir.cache-directory");
+    std::filesystem::create_directories(*cache_dir);
+    const auto current_time = std::time(nullptr);
+    std::string temp_file_name = *cache_dir + std::to_string(current_time);
+    auto temp_file = std::fstream(temp_file_name, std::ios_base::out);
+    if (!temp_file) {
+      return caf::make_error(ec::filesystem_error,
+                             fmt::format("failed to open temp file on "
+                                         "data load: {}",
+                                         detail::describe_errno()));
+    }
+    temp_file.write(reinterpret_cast<const char*>(serialized->data()),
+                    static_cast<std::streamsize>(serialized->size()));
+    if (!temp_file) {
+      return caf::make_error(ec::filesystem_error,
+                             fmt::format("failed write the temp file "
+                                         "on data load: {}",
+                                         detail::describe_errno()));
+    }
+    temp_file.close();
+    if (!temp_file) {
+      return caf::make_error(ec::filesystem_error,
+                             fmt::format("failed close the temp file: {}",
+                                         detail::describe_errno()));
+    }
+    auto mmdb = make_mmdb(temp_file_name);
```

Comment:
To guarantee that the temp file is still alive, it should only be closed after the call to `make_mmdb` so that there's always at least one reference to the file.